src/subdomains/core/history/mappers/transaction-dto.mapper.ts

```diff
@@ -183,6 +185,33 @@ export class TransactionDtoMapper {
       date: tx.created,
     };
   }
+
+  private static mapFees(entity: BuyCryptoExtended | BuyFiatExtended): FeeDto {
+    const referencePrice = entity.inputAmount / entity.inputReferenceAmount;
+
+    return {
+      rate: entity.percentFee,
+      fixed:
+        entity.absoluteFeeAmount &&
+        Util.roundReadable(entity.absoluteFeeAmount * referencePrice, isFiat(entity.inputAssetEntity)),
+      min:
+        entity.minFeeAmount &&
+        Util.roundReadable(entity.minFeeAmount * referencePrice, isFiat(entity.inputAssetEntity)),
+      network:
+        entity.blockchainFee &&
+        Util.roundReadable(entity.blockchainFee * referencePrice, isFiat(entity.inputAssetEntity)),
+      dfx:
+        entity.totalFeeAmount &&
+        entity.blockchainFee &&
+        Util.roundReadable(
+          (entity.totalFeeAmount - entity.blockchainFee) * referencePrice,
+          isFiat(entity.inputAssetEntity),
+        ),
+      total:
+        entity.totalFeeAmount &&
+        Util.roundReadable(entity.totalFeeAmount * referencePrice, isFiat(entity.inputAssetEntity)),
+    };
```

Comment:
Check for null (e.g. `entity.absoluteFeeAmount != null`) since we want to have 0, if it's 0